src/v/cluster/controller_backend.cc

```diff
@@ -863,6 +878,27 @@ ss::future<> controller_backend::remove_from_shard_table(
       });
 }
 
+ss::future<std::error_code> controller_backend::create_materialized(
+  model::ntp ntp, model::revision_id rev) {
+    auto cfg = _topics.local().get_topic_cfg(model::topic_namespace_view(ntp));
+    if (!cfg) {
+        // partition was already removed, do nothing
+        co_return errc::success;
+    }
+    if (likely(!_storage.local().log_mgr().get(ntp))) {
+        auto ntp_cfg = cfg->make_ntp_config(
+          _data_directory, ntp.tp.partition, rev);
+        co_await _storage.local().log_mgr().manage(std::move(ntp_cfg));
+    } else {
+        auto sr = _shard_table.local().shard_revision_for(ntp);
+        if (sr && sr->revision < rev) {
+            co_return errc::partition_already_exists;
+        }
+    }
+    co_await add_to_shard_table(std::move(ntp), ss::this_shard_id(), rev);
```

Comment:
but you do not check return result: is it same on each core

Reply:
@dotnwat this was addressed to work in the same way the other `add_to_shard_table` method does. A shard table method is invoked on all cores that returns no value. So theres nothing to really check. Either the insertion occurred or the item was already there so nothing happens. `controller_backend::create_partition` works the same way

Reply:
I do not know the right way, but looks like you need to assert that all cores have the same _shard_table.
You have 2 different branches and need to check result from all cores like there https://github.com/vectorizedio/redpanda/blob/5470a21ed76ade910e71dd3699c15619c7304606/src/v/cluster/security_manager.cc#L116